aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java

```diff
@@ -558,27 +563,20 @@ private static Artifact findArtifact(final ArtifactId id, final Collection<Artif
      * @return the artifact, which has been resolved.
      * @throws RuntimeException if the artifact can't be resolved
      */
-    @SuppressWarnings("deprecation")
     Artifact getOrResolveArtifact(final ArtifactId id) {
         Artifact result = this.artifactCache.get(id.toMvnId());
         if ( result == null ) {
             result = findArtifact(id, project.getAttachedArtifacts());
             if ( result == null ) {
-                result = findArtifact(id, project.getDependencyArtifacts());
+                result = findArtifact(id, project.getArtifacts());
```

Comment:
getDependencyArtifacts lists (afaik) only the direct dependencies - while getArtifacts is transitive depending on the phase it gets invoked in. Maybe we need to switch to getDependencyList

Reply:
I don't see a harm with also considering transitive dependencies, but maybe I am missing something. However, this is not related to the goal of this PR therefore I am reverting this change.